pkg/cache/context.go

```diff
@@ -315,7 +315,9 @@ func (ctx *Context) updateYuniKornPod(appID string, pod *v1.Pod) {
 
 	// treat terminated pods like a remove
 	if utils.IsPodTerminated(pod) {
-		ctx.notifyTaskComplete(appID, taskID)
+		if app != nil {
```

Comment:
Lets clean up a bit more: the line 310 we retrieve the app inside the if and we have a separate declaration of the var in 308. We can fold the two into one by retrieving the app outside the if:
```
app := ctx.getApplication(appID)
if app != nil {
```

Reply:
Done.